pyreal/visualize/feature_based_vis.py

```diff
@@ -366,9 +366,9 @@ def feature_scatter_plot(
     if legend_type == "discrete":
         plot_legend = True
 
-    if discrete is None:
-        discrete = not pd.api.types.is_numeric_dtype(values)
-    if discrete:
+    categorical = (discrete is None) and (not pd.api.types.is_numeric_dtype(values))
```

Comment:
This will set categorical to False is discrete=True, which is not correct behavior - categorical data is discrete. It also looks like we are no longer setting discrete when its passed in as None, which should not be the case